### PR TITLE
Add `GPTNeoXCausalLMPreprocessor`

### PR DESCRIPTION
**Pull request by shivance**
In favor of #1052

### CODE REVIEW

File: keras_nlp/models/gpt_neo_x/gpt_neo_x_preprocessor.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 """GPTNeoX preprocessor layer."""
 from keras_nlp.api_export import keras_nlp_export
-from keras_nlp.layers.start_end_packer import StartEndPacker
+from keras_nlp.layers import StartEndPacker
```
Comment (mattdangerw) [inline keras_nlp/models/gpt_neo_x/gpt_neo_x_preprocessor.py:17]: we should write this `from keras_nlp.layers.preprocessing.start_end_packer import StartEndPacker`. I can make this change as I merge.

**Review by mattdangerw — APPROVED** [review]
This looks good!

File: keras_nlp/models/gpt_neo_x/gpt_neo_x_preprocessor.py (hunk 1 above)
Comment (shivance) [reply keras_nlp/models/gpt_neo_x/gpt_neo_x_preprocessor.py]: Hey @mattdangerw made the change!

**Pull request merged**